python/lsst/meas/algorithms/astrometrySourceSelector.py

```diff
@@ -68,16 +72,24 @@ def selectSources(self, sourceCat, matches=None):
         """
         self._getSchemaKeys(sourceCat.schema)
 
-        result = table.SourceCatalog(sourceCat.table)
-        for source in sourceCat:
-            if self._isGood(source) and not self._isBad(source):
-                result.append(source)
+        if sourceCat.isContiguous():
+            bad = reduce(lambda x, y: np.logical_or(x, sourceCat.get(y)), self.config.badFlags, False)
+            good = self._isGood_vector(sourceCat)
+            result = sourceCat[good & ~bad]
+        else:
+            result = table.SourceCatalog(sourceCat.table)
+            for i, source in enumerate(sourceCat):
+                if self._isGood(source) and not self._isBad(source):
+                    result.append(source)
         return Struct(sourceCat=result)
 
     def _getSchemaKeys(self, schema):
         """Extract and save the necessary keys from schema with asKey."""
         self.parentKey = schema["parent"].asKey()
         self.nChildKey = schema["deblend_nChild"].asKey()
+        self.centroidXKey = schema["slot_Centroid_x"].asKey()
+        self.centroidYKey = schema["slot_Centroid_y"].asKey()
```

Comment:
I thought the field was "slot_Centroid" and it returned a tuple of the (y, x) centroid location. Does this also work?